src/Maestro/Maestro.Authentication/BarTokenAuthenticationHandler.cs

```diff
@@ -81,4 +113,17 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 
         return authHeader;
     }
+
+    private (int tokenId, string password)? DecodeToken(string input)
+    {
+        byte[] tokenBytes = WebEncoders.Base64UrlDecode(input);
+        if (tokenBytes.Length != PersonalAccessTokenUtilities.CalculateTokenSizeForPasswordSize(16))
+        {
+            return null;
+        }
+
+        int tokenId = IPAddress.NetworkToHostOrder(BitConverter.ToInt32(tokenBytes, 0));
```

Comment:
Why is this conversion needed here?

Reply:
I have no idea but it was taken from here: https://github.com/dotnet/dnceng-shared/blob/77fb751f40c92262450af10314e965a1a708857b/src/Microsoft.DotNet.Web.Authentication/AccessToken/PersonalAccessTokenAuthenticationHandler.cs#L87

Reply:
I guess its just a safety precaution:
![image](https://github.com/user-attachments/assets/4ee51422-8463-426f-863c-013681e05aac)